### DISCUSSION TITLE
how to configure GAL22V10 I/O/Q pin as input?

### DISCUSSION THREAD

**Discussion by lynchaj** [hneemann/Digital#1121] [Q&A] [answered]
I hope this question isn't a FAQ.  I've been searching the manual and online for help but can't seem to find an answer.

I have a logic schematic I would like to convert into a GAL22V10.  Unfortunately, the GAL22V10 supports only 10 pins as inputs and my logic schematic has 11 inputs.  It also has 5 outputs.  I thought (possibly misunderstood) that all 22 pins could be configured as inputs.  Or of the 22 pins, 10 could only be used as inputs and the rest could be inputs or outputs.  Maybe that's wrong?

10 of the 11 inputs are connected to input pins and the 1 remaining input is connected to pin 14.  The 5 outputs are connected to the usual output pins.  Digital is giving me an error message "Error during creation of hardware configuration.  Pin 14 is not an input."

Any help greatly appreciated.  Thank you.

**Comment by lynchaj** [comment, marked as answer]
[Z80CPU-GAL.zip](https://github.com/hneemann/Digital/files/10832504/Z80CPU-GAL.zip)
Hi
I am including a copy of the Digital logic schematic.  Thanks

**Reply by hneemann** [reply]
You are right: Ten pins are inputs and ten more pins can be configured as output or input.
Unfortunately the simulator does not support the configuration of the I/O pins as inputs. In the labs I have run, many outputs and only a few inputs were needed, so I would not have thought of using the I/Os as inputs. 
And with the move to the ATF1504 chips, further development of 22v10 support then essentially stopped.

**Comment by lynchaj** [comment]
Thank you for the reply.  I do appreciate your response.
It sounds like this is already a known issue.
Would it be worthwhile to open an issue on the Digital GitHub?  
Maybe someone will have an interest in adding this capability?
Thank you for developing this program. I really enjoy using it.
Thanks, Andrew Lynch


    On Sunday, February 26, 2023 at 02:32:54 PM EST, Helmut Neemann ***@***.***> wrote:  
 
 


You are right: Ten pins are inputs and ten more pins can be configured as output or input.
Unfortunately the simulator does not support the configuration of the I/O pins as inputs. In the labs I have run, many outputs and only a few inputs were needed, so I would not have thought of using the I/Os as inputs.
And with the move to the ATF1504 chips, further development of 22v10 support then essentially stopped.
